ortools/sat/csharp/CpModel.cs

```diff
@@ -1191,25 +1191,36 @@ internal LinearExpr GetLinearExpr<X>(X x)
 
     internal LinearExpressionProto GetLinearExpressionProto(LinearExpr expr, bool negate = false)
     {
-        Dictionary<IntVar, long> dict = new Dictionary<IntVar, long>();
-        long constant = LinearExpr.GetVarValueMap(expr, 1L, dict, terms_);
+        var dict = var_value_map_;
```

Comment:
why introduce a local variables ?

Reply:
locals generate slightly better code due to the fact that that the JIT is able to prove e.g. that dict is not null.